src/rdkafka_transport.c

```diff
@@ -503,6 +507,13 @@ void rd_kafka_transport_ssl_init (void) {
 	SSL_load_error_strings();
 	SSL_library_init();
 	OpenSSL_add_all_algorithms();
+#else
+#ifdef HAVE_OPENSSL_CRYPTO_THREADID_SET_CALLBACK
+	CRYPTO_THREADID_set_callback(rd_kafka_transport_ssl_threadid_cb);
```

Comment:
Can't find if this is actually needed on OpenSSL >=1.1.0, but other projects seem to not use it on that version, so maybe it should be moved to the < 1.1.0 block?

https://github.com/pycurl/pycurl/issues/491

Reply:
True. I moved it to the < 1.1.0 block.